File: module-app/src/main/java/maple/expectation/controller/dto/auth/LoginResponse.java
package maple.expectation.controller.dto.auth;

/**
 * 로그인 응답 DTO
 *
 * @param accessToken JWT Access Token
 * @param expiresIn Access Token 만료 시간 (초)
 * @param role 사용자 역할 (USER, ADMIN)
 * @param fingerprint API Key의 HMAC-SHA256 해시
 * @param refreshToken Refresh Token ID
 * @param refreshExpiresIn Refresh Token 만료 시간 (초)
 */
public record LoginResponse(
    String accessToken,
    Long expiresIn,
    String role,
    String fingerprint,
    String refreshToken,
    Long refreshExpiresIn) {

  /**
   * 정적 팩토리 메서드 (LoginResponse 생성)
   *
   * @param accessToken JWT Access Token
   * @param expiresIn Access Token 만료 시간 (초)
   * @param role 사용자 역할 (USER, ADMIN)
   * @param fingerprint API Key의 HMAC-SHA256 해시
   * @param refreshToken Refresh Token ID
   * @param refreshExpiresIn Refresh Token 만료 시간 (초)
   * @return LoginResponse 인스턴스
   */
  public static LoginResponse of(
      String accessToken,
      Long expiresIn,
      String role,
      String fingerprint,
      String refreshToken,
      Long refreshExpiresIn) {
    return new LoginResponse(
        accessToken, expiresIn, role, fingerprint, refreshToken, refreshExpiresIn);
  }
}

File: module-web/src/main/kotlin/maple/expectation/web/dto/LoginResponse.kt
package maple.expectation.web.dto

/**
 * Login response DTO (Issue #279: Refresh Token 추가)
 *
 * @param accessToken JWT access token
 * @param expiresIn Access Token expiration time (seconds)
 * @param user role
 * @param fingerprint account fingerprint (for Admin registration)
 * @param refreshToken Refresh Token ID (Issue #279)
 * @param refreshExpiresIn Refresh Token expiration time (seconds) (Issue #279)
 */
data class LoginResponse(
    val accessToken: String?,
    val expiresIn: Long?,
    val role: String?,
    val fingerprint: String?,
    val refreshToken: String? = null,
    val refreshExpiresIn: Long? = null,
) {
    // Safe accessors for Java tests (handle null from mocks)
    @JvmName("getAccessTokenSafe")
    fun getAccessToken(): String = accessToken ?: ""

    @JvmName("getExpiresInSafe")
    fun getExpiresIn(): Long = expiresIn ?: 0

    @JvmName("getRoleSafe")
    fun getRole(): String = role ?: ""

    @JvmName("getFingerprintSafe")
    fun getFingerprint(): String = fingerprint ?: ""

    @JvmName("getRefreshTokenSafe")
    fun getRefreshToken(): String = refreshToken ?: ""

    @JvmName("getRefreshExpiresInSafe")
    fun getRefreshExpiresIn(): Long = refreshExpiresIn ?: 0

    companion object {
        /**
         * Create LoginResponse with Refresh Token
         */
        @JvmStatic
        fun of(
            accessToken: String?,
            expiresIn: Long?,
            role: String?,
            fingerprint: String?,
            refreshToken: String?,
            refreshExpiresIn: Long?,
        ): LoginResponse = LoginResponse(
            accessToken,
            expiresIn,
            role,
            fingerprint,
            refreshToken,
            refreshExpiresIn,
        )

        /**
         * Create LoginResponse without Refresh Token (legacy compatibility)
         *
         * @deprecated Use [of] with all parameters instead
         */
        @Deprecated(
            "Use full constructor with refresh token",
            ReplaceWith("of(accessToken, expiresIn, role, fingerprint, null, null)"),
        )
        @JvmStatic
        fun of(accessToken: String?, expiresIn: Long?, role: String?, fingerprint: String?): LoginResponse =
            LoginResponse(accessToken, expiresIn, role, fingerprint)
    }
}
